services/apps/alcs/src/admin/unarchive-card/unarchive-card.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@nestjs/common';
+import { ApplicationService } from '../../application/application.service';
+import { CovenantService } from '../../covenant/covenant.service';
+import { ApplicationModificationService } from '../../decision/application-modification/application-modification.service';
+import { ApplicationReconsiderationService } from '../../decision/application-reconsideration/application-reconsideration.service';
+import { PlanningReviewService } from '../../planning-review/planning-review.service';
+
+@Injectable()
+export class UnarchiveCardService {
+  constructor(
+    private applicationService: ApplicationService,
+    private reconsiderationService: ApplicationReconsiderationService,
+    private planningReviewService: PlanningReviewService,
+    private modificationService: ApplicationModificationService,
+    private covenantService: CovenantService,
+  ) {}
+
+  async fetchByFileId(fileId: string) {
+    const result: {
+      cardUuid: string;
+      type: string;
+      status: string;
+      createdAt: number;
+    }[] = [];
+    const application = await this.applicationService.getDeletedCard(fileId);
```

Comment:
This looks like a good candidate to split into multiple functions. It should be easier later to keep track of all the different card types that get attached to the result list

Reply:
Done!